src/Catalog/PackageDownloader.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public class PackageDownloader
+    {
+        private const int BufferSize = 8192;
+
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+
+        public PackageDownloader(HttpClient httpClient, ILogger logger)
```

Comment:
This is a copy of `PackageDownloader` from NuGet.Jobs. I have removed some custom telemetry events and returned `null` on 404.